### PR TITLE
PayPal donations use paypal sdk to process subscription with smart buttons

### PR DESCRIPTION
**Pull request by ravinderk**
In favor of https://github.com/impress-org/give-recurring/issues/984

## Description
This PR adds support to Paypal smart buttons to process the subscription.

## Affects
This pr does not pr existing functionality.

## Pre-review Checklist
-   [x] Acceptance criteria satisfied and marked in a related issue
-   [x] Relevant `@since` tags included in DocBlocks
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

## Testing Instructions
Follow testing instructions mentioned in https://github.com/impress-org/give-recurring/pull/985

### CODE REVIEW

File: assets/src/js/frontend/paypal-commerce/SmartButtons.js [hunk 1]
```diff
@@ -21,36 +57,44 @@ class SmartButtons extends PaymentMethod {
...
 
 	/**
 	 * Render smart buttons.
 	 *
 	 * @since 2.9.0
+	 * @return {object} Return Promise
 	 */
 	renderPaymentMethodOption() {
 		this.smartButtonContainer = this.getButtonContainer();
+		const onCancelError = () => DonationForm.addErrors( this.jQueryForm, Give.form.fn.getErrorHTML( [ { message: givePayPalCommerce.defaultDonationCreationError } ] ) );
```
Comment (ravinderk) [inline assets/src/js/frontend/paypal-commerce/SmartButtons.js:83]: @JasonTheAdams What message we can show on the donation form when donor cancels the payment in PayPal popup. I am showing the default message currently.

Comment (JasonTheAdams) [reply assets/src/js/frontend/paypal-commerce/SmartButtons.js:83]: Don't show anything. If they canceled then there wasn't really any error.

Comment (ravinderk) [reply assets/src/js/frontend/paypal-commerce/SmartButtons.js]: fixed: https://github.com/impress-org/givewp/compare/d665df914e02...648d8fcb5e29

**Review by JasonTheAdams — APPROVED** [review]
Great work! As soon as the other PR is ready, let's merge this for testing!

**Pull request merged**